pkg/controller/compute/gke.go

```diff
@@ -193,6 +194,11 @@ func (r *Reconciler) _sync(instance *gcpcomputev1alpha2.GKECluster, client gke.C
 		return r.fail(instance, err)
 	}
 
+	if cluster.Status == gcpcomputev1alpha2.ClusterStateError {
+		instance.Status.State = gcpcomputev1alpha2.ClusterStateError
+		return r.fail(instance, fmt.Errorf(erroredClusterErrorMessageFormat, cluster.Status, cluster.StatusMessage))
```

Comment:
https://godoc.org/github.com/pkg/errors

I'd prefer to use `pkg/errors` (i.e. `errors.Errorf`) rather than `fmt.Errorf`.

Reply:
I think we should be updating the `Ready` condition, not the `Synced` (which is set by `r.fail`) condition here.

In theory there are no strict requirements around what conditions a managed resource must support, but in practice we support two: `Ready` and `Synced`. The former reflects the status of the resource under reconciliation, while the latter reflects the status of our attempts to reconcile it.

In this case I think we should be using `Unavailable` (a reason of the `Ready` condition) rather than `ReconcileError` (a reason of the `Synced` condition), because we are successfully reconciling the cluster (i.e. our reconcile logic is not encountering an error), but the cluster itself is in an error state according to its API.

Reply:
Thanks for the explanation about the difference between the two conditions.

That makes total sense. Opened the PR https://github.com/crossplaneio/crossplane-runtime/pull/44 which adds *UnavailableWithMessage* function to be consumed here.